Sources/VLCOneDriveTableViewController.m

```diff
@@ -96,78 +112,86 @@ - (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(N
-                if (url) {
-                    [mediaList addMedia:[VLCMedia mediaWithURL:url]];
-                    if (item.subtitleURL)
-                        [[mediaList mediaAtIndex:counter] addOptions:@{ kVLCSettingSubtitlesFilePath : item.subtitleURL }];
-                    counter ++;
-
-                    if (item == selectedObject)
-                        posIndex = mediaList.count - 1;
-                }
-            }
-
-            if (mediaList.count > 0)
-                [self streamMediaList:mediaList startingAtIndex:posIndex subtitlesFilePath:nil];
+            mediaList = [self createMediaListWithODItem:selectedItem positionIndex:&positionIndex];
         }
-    }
 
+        [self streamMediaList:mediaList startingAtIndex:positionIndex subtitlesFilePath:subtitlePath];
+    }
     [self.tableView deselectRowAtIndexPath:indexPath animated:NO];
 }
 
 - (void)streamMediaList:(VLCMediaList *)mediaList startingAtIndex:(NSInteger)startIndex subtitlesFilePath:(NSString *)subtitlesFilePath
 {
+    if (mediaList.count <= 0) {
+        NSLog(@"VLCOneDriveTableViewController: Empty or wrong mediaList");
+        return;
+    }
+
     VLCPlaybackController *vpc = [VLCPlaybackController sharedInstance];
     vpc.fullscreenSessionRequested = NO;
     [vpc playMediaList:mediaList firstIndex:startIndex subtitlesFilePath:subtitlesFilePath];
 }
 
-- (void)playAllAction:(id)sender
+- (VLCMediaList *)createMediaListWithODItem:(ODItem *)item positionIndex:(NSInteger *)index
 {
     NSUInteger counter = 0;
-    NSArray *folderItems = _oneDriveController.currentFolder.items;
+    NSArray *folderItems = _oneDriveController.currentListFiles;
     VLCMediaList *mediaList = [[VLCMediaList alloc] init];
-    for (VLCOneDriveObject *item in folderItems) {
-        if ((item.isFolder) || [item.name isSupportedSubtitleFormat])
+    for (ODItem *tmpItem in folderItems) {
+        if (tmpItem.folder || [tmpItem.name isSupportedSubtitleFormat]) {
             continue;
-        NSURL *url = [NSURL URLWithString:item.downloadPath];
+        }
+        NSURL *url = [NSURL URLWithString:tmpItem.dictionaryFromItem[@"@content.downloadUrl"]];
         if (url) {
             [mediaList addMedia:[VLCMedia mediaWithURL:url]];
-            if (item.subtitleURL)
-                [[mediaList mediaAtIndex:counter] addOptions:@{ kVLCSettingSubtitlesFilePath : item.subtitleURL }];
-            counter ++;
+            NSString *subtitlePath = [_oneDriveController configureSubtitleWithFileName:tmpItem.name
+                                                                            folderItems:folderItems];
+            if (subtitlePath) {
+                [[mediaList mediaAtIndex:counter] addOptions:@{ kVLCSettingSubtitlesFilePath : subtitlePath }];
+            }
+            // Index needed to know where to begin in the medialist
+            if (item == tmpItem) {
+                *index = mediaList.count - 1;
```

Comment:
*index ?

Reply:
yes.

Reply:
okay it looks a bit unintuitive but I can see the overall dependencies now